### DISCUSSION TITLE
Custom jobs

### DISCUSSION THREAD

**Discussion by ilyaChuk** [immich-app/immich#11246] [Q&A]
How can I make my own jobs? I want to do my own processing, which all files will go through. A type of script that will take turns accepting a file, making a condition that it is an image, not a video, and can return its modified version. Maybe even in a different format (recoding images to avi, to be precise). And so that it works when uploading each new file

I am a developer and I can get into the source files if necessary. But I would like something simpler.

**Comment by kaliavarun** [comment]
Hello, did you find any solution to this problem? I also have a use case where I would like to call a custom job sync/async on everyupload.

**Reply by timaschew** [reply]
Maybe this will work for you https://github.com/joojoooo/immich-upload-optimizer

**Comment by mheidinger** [comment]
Supporting this, I'd find it very useful to have the ability to develop / import custom jobs that can run on all photos and then incrementally on all new photos similar to the facial recognition. Usages I could see is e.g. tagging for detected screenshots, tagging based on file name (e.g. all whats app files), image optimization, etc.

**Comment by aguaviva** [comment]
A job I would like to have is to convert images to HEIC and bake in images the data from the database 
(geolocation, time, comments...)

**Comment by jd-1337** [comment]
I'd like to be able to do things like

- add description, tags, albums based on the image + metadata from other jobs (faces, location, ocr). Probably sent to self-hosted ai (or cloud)

- hard link the underlying file somewhere else on the file system based on tags, faces, location, albums, camera, exposure (any fixed things - something like a description could use a contains i guess)
